### DISCUSSION TITLE
FormScope如何控制Button是否启用？

### DISCUSSION THREAD

**Discussion by danatechgithub4** [junerver/ComposeHooks#23] [Q&A] [answered]
UseFormExample里面
```
            TButton(text = "submit", enabled = form.isValidated()) {
                toast(form.getAllFields().toString() + "isValidated :" + form.isValidated())
            }
```
代码`enabled = form.isValidated()` 没有任何效果。

**Comment by junerver** [comment]
没有效果符合预期，因为这个函数的返回值并非 `State` 状态，拿到状态值作为提交按钮的 `enable` 确实挺有用，明天我研究研究

**Reply by danatechgithub4** [reply]
是的，我现在暂时先分别监听几个Field，代码读起不美观。期待下个版本优化。

**Comment by junerver** [comment, marked as answer]
更新快照版本：2.0.1-SNAPSHOT
新增 `FormScope` 下的扩展函数 `FormInstance._isValidated(): State<Boolean>`，现在你可以在 `Form` 组件下使用它来获取表单的校验状态。
欢迎试用、反馈~
